rsocket-core/src/main/java/io/rsocket/RSocketResponder.java

```diff
@@ -488,7 +458,21 @@ private void handleChannel(int streamId, Payload payload, int initialRequestN) {
                 () -> sendProcessor.onNext(CancelFrameFlyweight.encode(allocator, streamId)))
             .doOnError(t -> handleError(streamId, t))
             .doOnRequest(
-                l -> sendProcessor.onNext(RequestNFrameFlyweight.encode(allocator, streamId, l)))
+                new LongConsumer() {
+                  boolean first = true;
+
+                  @Override
+                  public void accept(long l) {
+                    long n;
+                    if (first) {
+                      first = false;
+                      n = l - 1;
```

Comment:
just for the sake of readability (with a non-coding font), consider using either a different letter than `l` or the literal `1L` instead of `1`

Reply:
fixed